core/metacling/src/TClingCallbacks.cxx

```diff
@@ -919,7 +922,48 @@ void TClingCallbacks::UnlockCompilationDuringUserCodeExecution(void *StateInfo)
    TCling__UnlockCompilationDuringUserCodeExecution(StateInfo);
 }
 
-static bool shouldIgnore(llvm::StringRef FileName) {
+static bool shouldIgnore(const std::string& FileName,
+                         cling::DynamicLibraryManager* dyLibManager) {
+   if (llvm::sys::fs::is_directory(FileName))
```

Comment:
Out of curiosity, isn't that check also done in ```DynamicLibraryManager::isSharedLibrary```?

Reply:
We do not seem to have that. There we assume we passed a file.